#### Business Objective: What was the impact of the COVID-19 pandemic on the level of home court advantage?
- Question 1: How has the difference in win-loss ratio between home and away teams changed from the 2019-20 season to the 2022-23 season? (wl_home and wl_away columns in game table)
- Question 2: How has the difference in average points scored between home and away teams changed from the 2019-20 season to the 2022-23 season? (pts_home and pts_away columns in game table)
- Question 3: How has the difference in offensive and defensive rebounds between home and away teams changed from 2019-20 to 2022-23?(oreb_home, oreb_away, dreb_home, dreb_away in the game table)
- Question 4: How does the average number of three-point field goals made by home teams compare to that of away teams from the 2019-20 season to the 2022-23 season? (fg3_pct_home and fg3_pct_away columns in game table)
- Question 5: How does the free throw percentage of home teams compare to that of away teams from the 2019-20 season to the 2022-23 season? (ft_pct_home and ft_pct_away in the game table)

In [1]:
import pandas as pd
import sqlite3

con = sqlite3.connect("data/nba.sqlite")

seasons = ['2019', '2020', '2021', '2022']

Question 1

Intro: As the world embraced the shocks of a global pandemic in 2020, how did the basketball players adapt in a world with no fan presence on the court? Did the teams have no differences as they simply did as they've always done or were there noticable changes? Let's find out the difference between a pre pandemic league and a post pandemic one. 

Rising Action: First, we want to figure out how the teams actually did at home and away to see maybe what impact the difference of fan presence had on the home court advantage in the 2019-2020 season. The home team won 55.1% of the games while the away team won 44.8%. We can probably infer from this that there is a home court advantage in general but it isn't conclusive. 


Climax: Then, we ran the same analysis in the 2020-2021 and 2021-2022 seasons to see if this difference did exist... did the lack of presence of the fans on the home court have a huge impact? We figured out that the home team win-loss ratio was only at 54.3% for 2020-2021 and 54.4% for 2021-2022, and the away team win-loss ratio was only 45.6% in both season, there were changes in the pre pandemic world and a post pandemic one, the question was merely, was it enough to prove something?

Falling Action: Is there really such a huge difference between 55.1% vs 54.3%? Or 44.8% vs 45.6%? What do we even count as a huge difference? 

Conclusion: We believe that although there is some change, it isn't significant enough to prove that the pandemic had a huge impact on the home & away teams win-loss ratio, the home team was still outperforming despite having no fan presence in their home court.

In [2]:
covid_wl_home = pd.read_sql_query("SELECT team_name_home, wl_home, season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) ORDER BY season_id DESC", con)
covid_wl_home = covid_wl_home.replace('W', 1)
covid_wl_home = covid_wl_home.replace('L', 0)

covid_wl_home_group = covid_wl_home.groupby(['season_id']).agg({'wl_home': ['mean']})
covid_wl_home_group

C:\Users\tsidd\AppData\Local\Temp\ipykernel_36728\3688947812.py:3: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  covid_wl_home = covid_wl_home.replace('L', 0)


wl_home
               mean
season_id          
22019      0.551464
22020      0.543519
22021      0.543902
22022      0.580488

In [3]:
covid_wl_away = pd.read_sql_query("SELECT team_name_away, wl_away, season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) ORDER BY season_id DESC", con)
covid_wl_away = covid_wl_away.replace('W', 1)
covid_wl_away = covid_wl_away.replace('L', 0)

covid_wl_away_group = covid_wl_away.groupby(['season_id']).agg({'wl_away': ['mean']})
covid_wl_away_group

C:\Users\tsidd\AppData\Local\Temp\ipykernel_36728\734295079.py:3: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  covid_wl_away = covid_wl_away.replace('L', 0)


wl_away
               mean
season_id          
22019      0.448536
22020      0.456481
22021      0.456098
22022      0.419512

In [5]:
covid_wl_home = pd.read_sql_query("SELECT team_name_home, wl_home, season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) ORDER BY season_id DESC", con)
covid_wl_home = covid_wl_home.replace('W', 1)
covid_wl_home = covid_wl_home.replace('L', 0)

covid_wl_home_group = covid_wl_home.groupby(['team_name_home', 'season_id']).agg({'wl_home': ['mean']})
covid_wl_home_group

C:\Users\tsidd\AppData\Local\Temp\ipykernel_36728\1817048779.py:3: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  covid_wl_home = covid_wl_home.replace('L', 0)


wl_home
                                  mean
team_name_home     season_id          
Atlanta Hawks      22019      0.411765
                   22020      0.694444
                   22021      0.658537
                   22022      0.585366
Boston Celtics     22019      0.722222
...                                ...
Utah Jazz          22022      0.560976
Washington Wizards 22019      0.444444
                   22020      0.527778
                   22021      0.512195
                   22022      0.463415

[120 rows x 1 columns]

In [6]:
covid_wl_away = pd.read_sql_query("SELECT team_name_away, wl_away, season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) ORDER BY season_id DESC", con)
covid_wl_away = covid_wl_away.replace('W', 1)
covid_wl_away = covid_wl_away.replace('L', 0)

covid_wl_away_group = covid_wl_away.groupby(['team_name_away', 'season_id']).agg({'wl_away': ['mean']})
covid_wl_away_group

C:\Users\tsidd\AppData\Local\Temp\ipykernel_36728\2654544824.py:3: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  covid_wl_away = covid_wl_away.replace('L', 0)


wl_away
                                  mean
team_name_away     season_id          
Atlanta Hawks      22019      0.181818
                   22020      0.444444
                   22021      0.390244
                   22022      0.414634
Boston Celtics     22019      0.611111
...                                ...
Utah Jazz          22022      0.341463
Washington Wizards 22019      0.250000
                   22020      0.416667
                   22021      0.341463
                   22022      0.390244

[120 rows x 1 columns]

Question 2

Intro: As the world embraced the shocks of a global pandemic in 2020, how did the basketball players adapt in a world with no fan presence on the court? Did the teams have no differences as they simply did as they've always done or were there noticable changes? Let's find out the difference between a pre pandemic league and a post pandemic one. 

Rising Action: First, we want to figure out what the average points scored was by the teams at home and away to see the maybe what impact the difference of fan presence had on the home court advantage in the 2019-2020 season. The home team scored an average of 2.04 points more than the away teams, the home teams scored an average of 112.69 while the away teams scored an average of 110.65. We can probably infer from this that there is a home court advantage in general but it isn't conclusive. 

Climax: Then, we ran the same analysis in the 2020-2021 and 2021-2022 seasons to see how the average points scored changed... did the lack of presence of the fans on the home court have a huge impact? We figured out that the difference between the average points scored by home team vs the away team was 1.09 for the 2020-2021 season and 1.96 for the 2021-2022 season. There were some interesting changees in the pre pandemic world and a post pandemic one, the question was merely, was it enough to prove something?

Falling Action: The interesting aspect is the dip in 2020-2021 season which was when the bubble occurred with no fan presence and the rebound in the 2021-2022 season as it came back to 1.96 points! The dip and the rebound might turn out to be able to prove the difference in the home court advantage based on the environment. 

Conclusion: We believe that there could be some evidence for the impact of home court advantage based on the changes that occurred in 2020-2021 vs 2019-2020 & even 2021-2022, there is a fall off and recovery and this could've had an impact on the home court advantage.

In [7]:
result = []
for year in seasons:
    points_query = f"""
    SELECT 
        AVG(pts_home) AS avg_pts_home,
        AVG(pts_away) AS avg_pts_away
    FROM game
    WHERE SUBSTR(season_id, -4) = '{year}'
    """
    
    points_data = pd.read_sql_query(points_query, con)
    
    if not points_data.empty:
        avg_pts_home = points_data['avg_pts_home'][0]
        avg_pts_away = points_data['avg_pts_away'][0]
        
        points_diff = avg_pts_home - avg_pts_away
        
        result.append({
            'season_year': year,
            'avg_pts_home': avg_pts_home,
            'avg_pts_away': avg_pts_away,
            'points_diff': points_diff
        })

result_df = pd.DataFrame(result)
print(result_df)

  season_year  avg_pts_home  avg_pts_away  points_diff
0        2019    112.696413    110.651794     2.044619
1        2020    112.466283    111.369243     1.097039
2        2021    111.232659    109.271676     1.960983
3        2022    115.539683    112.931457     2.608225


In [8]:
covid_pts_home = pd.read_sql_query("SELECT team_name_home, AVG(pts_home), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_home, season_id", con)
covid_pts_home

team_name_home  AVG(pts_home) season_id
0         Atlanta Hawks     114.588235     22019
1         Atlanta Hawks     115.416667     22020
2         Atlanta Hawks     116.658537     22021
3         Atlanta Hawks     119.634146     22022
4        Boston Celtics     116.305556     22019
..                  ...            ...       ...
115           Utah Jazz     117.951220     22022
116  Washington Wizards     114.000000     22019
117  Washington Wizards     119.000000     22020
118  Washington Wizards     111.902439     22021
119  Washington Wizards     114.243902     22022

[120 rows x 3 columns]

In [10]:
covid_pts_home_19 = pd.read_sql_query("SELECT team_name_home, pts_home, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_home, season_id", con)
covid_pts_home_19['pts_home'].describe()

count     30.000000
mean     111.066667
std       14.971083
min       84.000000
25%      100.500000
50%      110.500000
75%      120.000000
max      158.000000
Name: pts_home, dtype: float64

In [9]:
covid_pts_home_21 = pd.read_sql_query("SELECT team_name_home, pts_home, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_home, season_id", con)
covid_pts_home_21['pts_home'].describe()

count     30.000000
mean     111.066667
std       12.940003
min       73.000000
25%      104.500000
50%      111.000000
75%      120.750000
max      138.000000
Name: pts_home, dtype: float64

In [10]:
covid_pts_away = pd.read_sql_query("SELECT team_name_away, AVG(pts_away), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_away, season_id", con)
covid_pts_away

team_name_away  AVG(pts_away) season_id
0         Atlanta Hawks     108.848485     22019
1         Atlanta Hawks     111.972222     22020
2         Atlanta Hawks     111.219512     22021
3         Atlanta Hawks     117.219512     22022
4        Boston Celtics     111.000000     22019
..                  ...            ...       ...
115           Utah Jazz     116.195122     22022
116  Washington Wizards     114.833333     22019
117  Washington Wizards     114.277778     22020
118  Washington Wizards     105.341463     22021
119  Washington Wizards     112.073171     22022

[120 rows x 3 columns]

In [11]:
covid_pts_away_19 = pd.read_sql_query("SELECT team_name_away, pts_away, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_away, season_id", con)
covid_pts_away_19['pts_away'].describe()

count     30.00000
mean     111.00000
std       17.15246
min       85.00000
25%       99.00000
50%      107.50000
75%      122.00000
max      159.00000
Name: pts_away, dtype: float64

In [12]:
covid_pts_away_21 = pd.read_sql_query("SELECT team_name_away, pts_away, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_away, season_id", con)
covid_pts_away_21['pts_away'].describe()

count     30.000000
mean     115.266667
std        8.893753
min       99.000000
25%      107.500000
50%      115.000000
75%      122.500000
max      131.000000
Name: pts_away, dtype: float64

Question 3

Intro: As the world embraced the shocks of a global pandemic in 2020, how did the basketball players adapt in a world with no fan presence on the court? Did the teams have no differences as they simply did as they've always done or were there noticable changes? Let's find out the difference between a pre pandemic league and a post pandemic one. 

Rising Action: First, we want to figure out what the teams averaged on their defensive and offensive rebounds at home and away to see maybe what impact the difference of fan presence had on the home court advantage in the 2019-2020 season. In the 2019-2020 season, home teams had an average of .38 more offensive rebounds and .95 defensive rebounds compared to the away team. In both the offensive and defensive rebounds, the home team did outperform the away team.

Climax: Then, we ran the same analysis in the 2020-2021 and 2021-2022 seasons to see how the offensive and defensive rebounds at home and away changed... did the lack of presence of the fans on the home court have a huge impact? We figured out that in the 2020-2021 season, the home team scored .15 more offensive rebounds and .68 more defensive rebounds, and in 2021-2022 season, the home team had .07 more offensive rebounds and .54 defensive rebounds, it seems to be going down over time. 

Falling Action: As the fans came bacin the 2021-2022 season, the offensive and defensive rebounds from the team only further decreased and this trend stayed on the downtrend going into 2022-2023 as well...

Conclusion: Although we cannot conclude any significance of the impact, we can see that it's a constant downtrend, regardless of fans being back on the court or not and the home court advantage is eroding over time.

In [13]:
result = []
for year in seasons:
    rebounds_query = f"""
    SELECT 
        AVG(oreb_home) AS avg_oreb_home,
        AVG(oreb_away) AS avg_oreb_away,
        AVG(dreb_home) AS avg_dreb_home,
        AVG(dreb_away) AS avg_dreb_away
    FROM game
    WHERE SUBSTR(season_id, -4) = '{year}'
    """
    
    rebounds_data = pd.read_sql_query(rebounds_query, con)
    
    if not rebounds_data.empty:
        avg_oreb_home = rebounds_data['avg_oreb_home'][0]
        avg_oreb_away = rebounds_data['avg_oreb_away'][0]
        avg_dreb_home = rebounds_data['avg_dreb_home'][0]
        avg_dreb_away = rebounds_data['avg_dreb_away'][0]
        
        oreb_diff = avg_oreb_home - avg_oreb_away
        dreb_diff = avg_dreb_home - avg_dreb_away
        
        result.append({
            'season_year': year,
            'avg_oreb_home': avg_oreb_home,
            'avg_oreb_away': avg_oreb_away,
            'oreb_diff': oreb_diff,
            'avg_dreb_home': avg_dreb_home,
            'avg_dreb_away': avg_dreb_away,
            'dreb_diff': dreb_diff
        })

result_df = pd.DataFrame(result)
print(result_df)

  season_year  avg_oreb_home  avg_oreb_away  oreb_diff  avg_dreb_home  \
0        2019      10.179353       9.794401   0.384952      35.209974   
1        2020       9.896382       9.742599   0.153783      34.866776   
2        2021      10.309249      10.236994   0.072254      34.416185   
3        2022      10.485570      10.419913   0.065657      33.575758   

   avg_dreb_away  dreb_diff  
0      34.256343   0.953631  
1      34.178454   0.688322  
2      33.875000   0.541185  
3      32.632756   0.943001  


In [14]:
covid_reb_home = pd.read_sql_query("SELECT team_name_home, AVG(oreb_home), AVG(dreb_home), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_home, season_id", con)
covid_reb_home

team_name_home  AVG(oreb_home)  AVG(dreb_home) season_id
0         Atlanta Hawks       10.205882       34.323529     22019
1         Atlanta Hawks       10.388889       36.527778     22020
2         Atlanta Hawks        9.341463       34.268293     22021
3         Atlanta Hawks       11.536585       34.439024     22022
4        Boston Celtics       10.138889       35.194444     22019
..                  ...             ...             ...       ...
115           Utah Jazz       12.000000       35.121951     22022
116  Washington Wizards       10.277778       32.861111     22019
117  Washington Wizards        9.611111       35.944444     22020
118  Washington Wizards        9.024390       34.512195     22021
119  Washington Wizards        9.317073       34.829268     22022

[120 rows x 4 columns]

In [15]:
covid_reb_home = pd.read_sql_query("SELECT team_name_home, oreb_home, dreb_home, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_home, season_id", con)
covid_reb_home['oreb_home'].describe()

count    30.000000
mean     11.133333
std       3.401149
min       5.000000
25%       9.000000
50%      11.000000
75%      12.750000
max      21.000000
Name: oreb_home, dtype: float64

In [16]:
covid_reb_home = pd.read_sql_query("SELECT team_name_home, oreb_home, dreb_home, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_home, season_id", con)
covid_reb_home['oreb_home'].describe()

count    30.000000
mean      9.700000
std       3.602202
min       3.000000
25%       7.000000
50%      10.000000
75%      12.750000
max      19.000000
Name: oreb_home, dtype: float64

In [17]:
covid_reb_home = pd.read_sql_query("SELECT team_name_home, oreb_home, dreb_home, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_home, season_id", con)
covid_reb_home['dreb_home'].describe()

count    30.000000
mean     36.533333
std       5.763939
min      26.000000
25%      32.000000
50%      36.500000
75%      41.000000
max      50.000000
Name: dreb_home, dtype: float64

In [18]:
covid_reb_home = pd.read_sql_query("SELECT team_name_home, oreb_home, dreb_home, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_home, season_id", con)
covid_reb_home['dreb_home'].describe()

count    30.000000
mean     36.166667
std       5.866817
min      26.000000
25%      31.000000
50%      36.500000
75%      40.000000
max      49.000000
Name: dreb_home, dtype: float64

In [19]:
covid_reb_away = pd.read_sql_query("SELECT team_name_away, AVG(oreb_away), AVG(dreb_away), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_away, season_id", con)
covid_reb_away

team_name_away  AVG(oreb_away)  AVG(dreb_away) season_id
0         Atlanta Hawks        9.515152       32.424242     22019
1         Atlanta Hawks       10.722222       33.611111     22020
2         Atlanta Hawks       10.731707       33.609756     22021
3         Atlanta Hawks       10.902439       31.878049     22022
4        Boston Celtics       11.194444       35.611111     22019
..                  ...             ...             ...       ...
115           Utah Jazz       11.609756       33.024390     22022
116  Washington Wizards       10.055556       30.888889     22019
117  Washington Wizards        9.750000       35.083333     22020
118  Washington Wizards        8.951220       33.731707     22021
119  Washington Wizards        9.560976       33.560976     22022

[120 rows x 4 columns]

In [20]:
covid_reb_away = pd.read_sql_query("SELECT team_name_away, oreb_away, dreb_away, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_away, season_id", con)
covid_reb_away['oreb_away'].describe()

count    30.000000
mean      9.833333
std       4.291799
min       3.000000
25%       6.250000
50%      10.000000
75%      12.000000
max      20.000000
Name: oreb_away, dtype: float64

In [21]:
covid_reb_away = pd.read_sql_query("SELECT team_name_away, oreb_away, dreb_away, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_away, season_id", con)
covid_reb_away['oreb_away'].describe()

count    30.000000
mean      9.133333
std       3.115405
min       5.000000
25%       7.000000
50%       8.500000
75%      11.000000
max      18.000000
Name: oreb_away, dtype: float64

In [22]:
covid_reb_away = pd.read_sql_query("SELECT team_name_away, oreb_away, dreb_away, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_away, season_id", con)
covid_reb_away['dreb_away'].describe()

count    30.000000
mean     35.766667
std       5.624352
min      25.000000
25%      32.000000
50%      35.000000
75%      40.000000
max      47.000000
Name: dreb_away, dtype: float64

In [23]:
covid_reb_away = pd.read_sql_query("SELECT team_name_away, oreb_away, dreb_away, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_away, season_id", con)
covid_reb_away['dreb_away'].describe()

count    30.000000
mean     36.166667
std       5.180090
min      24.000000
25%      33.250000
50%      36.000000
75%      40.000000
max      47.000000
Name: dreb_away, dtype: float64

Question 4

Intro: As the world embraced the shocks of a global pandemic in 2020, how did the basketball players adapt in a world with no fan presence on the court? Did the teams have no differences as they simply did as they've always done or were there noticable changes? Let's find out the difference between a pre pandemic league and a post pandemic one. 

Rising action: First, we want to figure out what the 3 point field goal percentages were from the teams at home and away to see maybe what impact the difference of fan presence had on the home court advantage in the 2019-2020 season. In the 2019-2020 season, home teams scored 36.14% as opposed to 35.4% from the away teams, with a difference of .74% in their 3 point shooting percentages.

Climax: Climax: Then, we ran the same analysis in the 2020-2021 and 2021-2022 seasons to see how 3 point percentages at home and away changed... did the lack of presence of the fans on the home court have a huge impact? In the 2020-2021 season, home teams scored 36.7% of their 3 points as opposed to 36.2% by the away team, the difference being .5% and in the 2021-2022 season, home teams scored 35.4% of their 3 pointers as opposed to the 35.0% by the away team, with the difference being only .4%. We are seeing a downtrend.

Falling action: We notice a downtrend in the post-pandemic world with no fans but in the 2022-2023 season, the 3 point percentages almost return back to the values in 2019-2020, like a reversion. 

Conclusion: Although, we cannot conclude a significance on the impact of home court advantages due to the 3 point percentages, we can still see some variance that occurs in a pre-pandemic home court with fans vs a post-pandemic court with no fans and then how it reversed in the post-pandemic court with fans again in the 2022-2023 season.

In [24]:
result = []
for year in seasons:
    fg3_query = f"""
    SELECT 
        AVG(fg3_pct_home) AS avg_fg3_pct_home,
        AVG(fg3_pct_away) AS avg_fg3_pct_away
    FROM game
    WHERE SUBSTR(season_id, -4) = '{year}'
    """
    
    fg3_data = pd.read_sql_query(fg3_query, con)
    
    if not fg3_data.empty:
        avg_fg3_pct_home = fg3_data['avg_fg3_pct_home'][0]
        avg_fg3_pct_away = fg3_data['avg_fg3_pct_away'][0]
        
        result.append({
            'season_year': year,
            'avg_fg3_pct_home': avg_fg3_pct_home,
            'avg_fg3_pct_away': avg_fg3_pct_away
        })

result_df = pd.DataFrame(result)
print(result_df)

  season_year  avg_fg3_pct_home  avg_fg3_pct_away
0        2019          0.361411          0.353990
1        2020          0.367235          0.362348
2        2021          0.354915          0.350280
3        2022          0.362962          0.351737


In [25]:
covid_fg3_home = pd.read_sql_query("SELECT team_name_home, AVG(fg3_pct_home), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_home, season_id", con)
covid_fg3_home

team_name_home  AVG(fg3_pct_home) season_id
0         Atlanta Hawks           0.344735     22019
1         Atlanta Hawks           0.381444     22020
2         Atlanta Hawks           0.380049     22021
3         Atlanta Hawks           0.347439     22022
4        Boston Celtics           0.378417     22019
..                  ...                ...       ...
115           Utah Jazz           0.363171     22022
116  Washington Wizards           0.371778     22019
117  Washington Wizards           0.346611     22020
118  Washington Wizards           0.341512     22021
119  Washington Wizards           0.370707     22022

[120 rows x 3 columns]

In [26]:
covid_fg3_home = pd.read_sql_query("SELECT team_name_home, fg3_pct_home, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_home, season_id", con)
covid_fg3_home['fg3_pct_home'].describe()

count    30.000000
mean      0.332833
std       0.077337
min       0.217000
25%       0.274750
50%       0.333000
75%       0.356500
max       0.556000
Name: fg3_pct_home, dtype: float64

In [27]:
covid_fg3_home = pd.read_sql_query("SELECT team_name_home, fg3_pct_home, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_home, season_id", con)
covid_fg3_home['fg3_pct_home'].describe()

count    30.00000
mean      0.33530
std       0.09684
min       0.12100
25%       0.27600
50%       0.35350
75%       0.40325
max       0.55600
Name: fg3_pct_home, dtype: float64

In [28]:
covid_fg3_away = pd.read_sql_query("SELECT team_name_away, AVG(fg3_pct_away), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_away, season_id", con)
covid_fg3_away

team_name_away  AVG(fg3_pct_away) season_id
0         Atlanta Hawks           0.319788     22019
1         Atlanta Hawks           0.358944     22020
2         Atlanta Hawks           0.363439     22021
3         Atlanta Hawks           0.359341     22022
4        Boston Celtics           0.353278     22019
..                  ...                ...       ...
115           Utah Jazz           0.340805     22022
116  Washington Wizards           0.365111     22019
117  Washington Wizards           0.359667     22020
118  Washington Wizards           0.344366     22021
119  Washington Wizards           0.337756     22022

[120 rows x 3 columns]

In [29]:
covid_fg3_away = pd.read_sql_query("SELECT team_name_away, fg3_pct_away, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_away, season_id", con)
covid_fg3_away['fg3_pct_away'].describe()

count    30.000000
mean      0.344267
std       0.099422
min       0.152000
25%       0.289500
50%       0.346500
75%       0.379000
max       0.563000
Name: fg3_pct_away, dtype: float64

In [30]:
covid_fg3_away = pd.read_sql_query("SELECT team_name_away, fg3_pct_away, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_away, season_id", con)
covid_fg3_away['fg3_pct_away'].describe()

count    30.000000
mean      0.373267
std       0.074224
min       0.229000
25%       0.345250
50%       0.365000
75%       0.412750
max       0.538000
Name: fg3_pct_away, dtype: float64

Question 5

Intro: As the world embraced the shocks of a global pandemic in 2020, how did the basketball players adapt in a world with no fan presence on the court? Did the teams have no differences as they simply did as they've always done or were there noticable changes? Let's find out the difference between a pre pandemic league and a post pandemic one. 

Rising Action: First, we want to figure out what the free throw percentages were from the teams at home and away to see maybe what impact the difference of fan presence had on the home court advantage in the 2019-2020 season. In the 2019-2020 season, away teams actually had a higher free throw percentages, with the figures being 77.24% for the away team and 77.18% for the home team. 

Climax: Then, we ran the same analysis in the 2020-2021 and 2021-2022 seasons to see how the free throw percentages at home and away changed... did the lack of presence of the fans on the home court have a huge impact? In the 2020-2021 season, we can actually see the home team gaining advantage back from the away team with a difference of .4% between the home and away team. In the 2021-2022 season, the edge swung back to the away team with the difference of .08% between away and the home teams. These numbers are far smaller than the number pre-pandemic.

Falling Action: The trend once again reversed to the home team in the 2022-2023 season, with them leading by an anverage of .02% in the free throw made percentages, there is too much volatility in the form that it's a constant back and forth every year and not by huge amounts of difference.

Conclusion: There is just not huge differences nor trend to be proven that the free throws made percentages had a huge impact on the home court advantage pre and post pandemic.

In [31]:
result = []
for year in seasons:
    ft_query = f"""
    SELECT 
        AVG(ft_pct_home) AS avg_ft_pct_home,
        AVG(ft_pct_away) AS avg_ft_pct_away
    FROM game
    WHERE SUBSTR(season_id, -4) = '{year}'
    """
    
    ft_data = pd.read_sql_query(ft_query, con)
    
    if not ft_data.empty:
        avg_ft_pct_home = ft_data['avg_ft_pct_home'][0]
        avg_ft_pct_away = ft_data['avg_ft_pct_away'][0]
        
        result.append({
            'season_year': year,
            'avg_ft_pct_home': avg_ft_pct_home,
            'avg_ft_pct_away': avg_ft_pct_away
        })

result_df = pd.DataFrame(result)
print(result_df)

  season_year  avg_ft_pct_home  avg_ft_pct_away
0        2019         0.771852         0.772411
1        2020         0.779016         0.775290
2        2021         0.775135         0.775920
3        2022         0.782598         0.780615


In [32]:
covid_fg_home = pd.read_sql_query("SELECT team_name_home, AVG(fg_pct_home), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_home, season_id", con)
covid_fg_home

team_name_home  AVG(fg_pct_home) season_id
0         Atlanta Hawks          0.459618     22019
1         Atlanta Hawks          0.477889     22020
2         Atlanta Hawks          0.482902     22021
3         Atlanta Hawks          0.482561     22022
4        Boston Celtics          0.469833     22019
..                  ...               ...       ...
115           Utah Jazz          0.471341     22022
116  Washington Wizards          0.457583     22019
117  Washington Wizards          0.481972     22020
118  Washington Wizards          0.479000     22021
119  Washington Wizards          0.486415     22022

[120 rows x 3 columns]

In [33]:
covid_fg_home = pd.read_sql_query("SELECT team_name_home, fg_pct_home, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_home, season_id", con)
covid_fg_home['fg_pct_home'].describe()

count    30.000000
mean      0.441867
std       0.061571
min       0.299000
25%       0.400500
50%       0.432500
75%       0.474000
max       0.626000
Name: fg_pct_home, dtype: float64

In [34]:
covid_fg_home = pd.read_sql_query("SELECT team_name_home, fg_pct_home, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_home, season_id", con)
covid_fg_home['fg_pct_home'].describe()

count    30.000000
mean      0.451600
std       0.052411
min       0.342000
25%       0.403500
50%       0.467000
75%       0.483250
max       0.556000
Name: fg_pct_home, dtype: float64

In [35]:
covid_fg_away = pd.read_sql_query("SELECT team_name_away, AVG(fg_pct_away), season_id FROM game WHERE season_id IN (22022, 22021, 22020, 22019) GROUP BY team_name_away, season_id", con)
covid_fg_away

team_name_away  AVG(fg_pct_away) season_id
0         Atlanta Hawks          0.440697     22019
1         Atlanta Hawks          0.460389     22020
2         Atlanta Hawks          0.459366     22021
3         Atlanta Hawks          0.486098     22022
4        Boston Celtics          0.454000     22019
..                  ...               ...       ...
115           Utah Jazz          0.476951     22022
116  Washington Wizards          0.458611     22019
117  Washington Wizards          0.467056     22020
118  Washington Wizards          0.465390     22021
119  Washington Wizards          0.487366     22022

[120 rows x 3 columns]

In [36]:
covid_fg_away = pd.read_sql_query("SELECT team_name_away, fg_pct_away, season_id FROM game WHERE season_id IN (22019) GROUP BY team_name_away, season_id", con)
covid_fg_away['fg_pct_away'].describe()

count    30.000000
mean      0.442967
std       0.063498
min       0.326000
25%       0.395750
50%       0.435500
75%       0.472500
max       0.625000
Name: fg_pct_away, dtype: float64

In [37]:
covid_fg_away = pd.read_sql_query("SELECT team_name_away, fg_pct_away, season_id FROM game WHERE season_id IN (22020) GROUP BY team_name_away, season_id", con)
covid_fg_away['fg_pct_away'].describe()

count    30.000000
mean      0.481567
std       0.045188
min       0.374000
25%       0.455500
50%       0.478500
75%       0.509750
max       0.561000
Name: fg_pct_away, dtype: float64

In [38]:
con.close()